### PR TITLE
USWDS - Header: Enable chevrons and add/remove icons to be visible in WHC/Forced Colors mode. Resolves #4453

### LINKED ISSUE #4453: Chevron on primary menu items may not meet contrast criteria in Windows high contrast/forced colors mode
### Describe the bug

The chevron on individual items in the mega menu may not meet color contrast criteria when using Window's High Contrast Mode. 

Windows high contrast lets you select multiple themes (black on white, white on black) or create custom themes. The chevron's background color does not respond to the theme changing colors. If you have a white on black theme (the default setting) the color of the icon is barely distinguishable from the background color. 

### Steps to reproduce the bug

1. Turn on Windows High Contrast Mode.
2. Ensure you are using a white on black theme.
2. Go to the [Header](https://designsystem.digital.gov/components/header/) component on the USWDS website. 
3. Click on the [Basic Header](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/components/preview/header--basic.html). 
4. Click on the component preview.
5. You will be unable to see the chevron (indicating a drop down menu) unless you hover your mouse over it. 

### Expected behavior
Given that this chevron conveys visual meaning that aids user navigation, it should be visible not matter the chosen theme. 

### Screenshots

Screen Shot 2022-01-07 at 3 38 06 PM

### System setup
- **USWDS version:**
- **Device: MacBook Pro**
- **Operating system: Windows 10 (Parallels), Big Sur 11.6**
- **Browser and version: Edge**

### Additional context

We/I will be filing a PR for this issue next week. Just wanted to add the issue here for tracking purposes.

### PR DESCRIPTION
**Pull request by mahoneycm**
## Preview

[Headers →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/cm-chevron-mask-mixin/components/detail/header--default.html)

## Description

Resolves #4453.

This PR uses the place-icon mixin to create a CSS mask for the Header navigation icons and enables them to be visible in WHC using  `@media (forced-colors: active) {background-color: ButtonText;}`

## Additional information

Currently:
image

After Fix:
image

Bonus Mobile view:
image

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [x] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `npm test` and make sure the tests for the files you have changed have passed.
- [ ] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [x] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

**Comment by mahoneycm** [comment]
I noticed the close button was disappearing on mobile while in WHC. Using the same solution as the chevron, I was able to replace the inline svg with  `place-icon` and add the forced colors media query to remedy this!

---- 

Before:

image


After:

image

**Review by thisisdano — APPROVED** [review]
Nice work!

**Review by mejiaj — CHANGES_REQUESTED** [review]
Looks and works pretty well. Thank you!

Some additional improvements we could make:
- Add some HCM styles to the navigation bar [[_add-bar.scss]](https://github.com/uswds/uswds/blob/11e947cc1aa4875c4c406ff203bfbc1ac8b11bb1/src/stylesheets/core/mixins/_add-bar.scss#L18) (used in `usa-current` here and in sidenav)
- Add a background or border to the search button for HCM only

File: src/stylesheets/components/_navigation.scss [hunk 1]
```diff
@@ -11,6 +11,69 @@ $sliding-panel-width: "card-lg";
...
+    "name": "expand_less",
+  )
+);
+
+// Maps for place-icon mixin
+$contrast-bg: "default";
+$after: "after";
+$before: "before";
+$icon-margin: "auto";
+$vertical-align: "auto";
```
Comment (amyleadem) [inline src/stylesheets/components/_navigation.scss:75]: I don't believe`vertical-margin: auto` has any meaning. Do you mean `middle`?

Reference: https://developer.mozilla.org/en-US/docs/Web/CSS/vertical-align

**Comment by amyleadem** [comment]
@mahoneycm 
Visual note: In standard mode, there is a difference in color for the expand icons. `develop` currently shows the icons as black, in this PR the col.ors are gray and blue.
image

**Comment by mahoneycm** [comment]
Closed in favor of #4604

**Pull request closed without merging**